Sources/muterCore/CLICommands/RunCommand/Steps/DiscoverSourceFiles.swift

```diff
@@ -48,40 +47,54 @@ private extension DiscoverSourceFiles {
         ]
     }
     
-    func discoverSourceFiles(inDirectoryAt path: FilePath,
-                             excludingPathsIn providedExcludeList: [FilePath] = []) -> [FilePath] {
+    private func discoverSourceFiles(
+        inDirectoryAt rootPath: FilePath,
+        excludingPathsIn providedExcludeList: [FilePath] = []
+    ) -> [FilePath] {
         let excludeList = providedExcludeList + defaultExcludeList
-        let subpaths = fileManager.subpaths(atPath: path) ?? []
+        let subpaths = fileManager.subpaths(atPath: rootPath) ?? []
+        
         return includeSwiftFiles(
             from: subpaths
-                .exclude(pathsContainingItems(from: excludeList))
-                .map(curry(append)(path))
+                .exclude(pathsContainingItems(root: rootPath, from: excludeList))
+                .map { append(root: rootPath, to: $0) }
         )
     }
     
-    func pathsContainingItems(from excludeList: [FilePath]) -> (FilePath) -> Bool {
-        return { (path: FilePath) in
-            
-            for item in excludeList where path.contains(item) {
+    private func pathsContainingItems(root: FilePath, from excludeList: [FilePath]) -> (FilePath) -> Bool {
```

Comment:
I think this would read better with the parameter order reversed. `pathsContainingItems(from:root:)`. But I don't fully understand what it does yet; a doc comment explaining the parameters would be great.